### ISSUE TITLE
Verification doesn't work when using public key encryption

### ISSUE THREAD

**Issue by taybin** [ahopkins/sanic-jwt#49]
With the master branch:

I'm using the RS256 algorithm, but I expect this will fail when using any of the other asymmetrical algorithms.  When accessing `/auth/verify` with the token, I get the failure:

```
  File "/usr/local/lib/python3.6/site-packages/sanic/app.py", line 556, in handle_request
    response = await response
  File "/usr/local/lib/python3.6/site-packages/sanic_jwt/blueprint.py", line 102, in verify
    request, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/sanic_jwt/authentication.py", line 169, in verify
    payload = self._decode(token, verify=verify)
  File "/usr/local/lib/python3.6/site-packages/sanic_jwt/authentication.py", line 62, in _decode
    **kwargs
  File "/usr/local/lib/python3.6/site-packages/jwt/api_jwt.py", line 78, in decode
    jwt, key=key, algorithms=algorithms, options=options, **kwargs
  File "/usr/local/lib/python3.6/site-packages/jwt/api_jws.py", line 140, in decode
    key, algorithms)
  File "/usr/local/lib/python3.6/site-packages/jwt/api_jws.py", line 206, in _verify_signature
    if not alg_obj.verify(signing_input, key, signature):
  File "/usr/local/lib/python3.6/site-packages/jwt/algorithms.py", line 317, in verify
    key.verify(sig, msg, padding.PKCS1v15(), self.hash_alg())
AttributeError: '_RSAPrivateKey' object has no attribute 'verify'
```

From my reading, this is because you need to use the public key to verify the token, since it was signed with the private key.  But sanic-jwt doesn't have a way to set the public-key.  It just has SANIC_JWT_SECRET.

https://github.com/jpadilla/pyjwt/issues/81

**Comment by ahopkins** [comment]
Thanks @taybin for the heads up. I have not myself tested this with a hashing algorithm that requires both a public and private key, so there is definitely no logic in there now. This is definitely on the short list for fixing ASAP. 

In the meanwhile, if you or anyone else out there wants to take a stab at this, it would be a good place to add a PR.

**Comment by vltr** [comment]
I thought of fixing this by creating a config key for private keys, where the secret could also the public key and the other (private) key valid only for signing. Also, they can be strings (the content of the key itself or a path to the key, using `os.path.isfile`) or a [`Path`](https://docs.python.org/3/library/pathlib.html#pathlib.Path) object.

Checking for it's usage (if RS* or EC*) would be a must IMO. Let me know your thoughts.

**Comment by ahopkins** [comment]
@vltr I think you volunteered for this one ;-)

I certainly agree on the string and path to file. That makes sense. I am not sure I understand the proposal though.

Under the hood, we are using [PyJWT](https://pyjwt.readthedocs.io/en/latest/algorithms.html). [This is the functionality we are missing](https://pyjwt.readthedocs.io/en/latest/usage.html#encoding-decoding-tokens-with-rs256-rsa).

**Comment by vltr** [comment]
I'm always volunteering! :joy:

The proposal is simple: do not break backward compatibility and enable the usage of certificates (from a string or reading from the filesystem) to sign (requiring a private key) and validate (requiring a public key) the tokens when using asymmetric cryptography. The first link from PyJWT explains this :wink:

**Comment by ahopkins** [comment]
Got it. 👍

Adam Hopkins

**Comment by vltr** [comment]
@taybin this bug is now fixed, but will only be available when version 1.0 hits the street :wink:

**Comment by vltr** [comment]
Fixed in d6ce8458dccad3d1f17c5e1dbb3dc82c79cbb607 for #58
